### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by yuya-oc**
Before submitting, please confirm you've
 - [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

Please provide the following information:

**Summary**
Update dependencies and devDependencies.

React 15.5 has some deprecation. https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html
In this PR, I used drop-in replacements, `prop-types`, `create-react-class` and `react-transition-group`.
So we need to review NOTICE.txt.

**Issue link**
#532 

**Test Cases**
Installation and normal usage.

For UI part, CircleCI would do automated tests.

**Additional Notes**
https://circleci.com/gh/yuya-oc/desktop/253#artifacts

### CODE REVIEW

**Comment by jasonblais** [comment]
Awesome! I've downloaded the artifact and will test it out this week. 

I'll also review NOTICE.txt, thanks for preparing it

**Comment by jasonblais** [comment]
I've been testing this out and have no issues. The only one that started appearing is #537 but it's not related to this PR.

Have also reviewed NOTICE.txt and looks good, thanks Yuya!

File: package.json [hunk 1]
```diff
@@ -38,27 +38,27 @@
...
-    "babel-preset-react": "^6.23.0",
+    "babel-preset-react": "^6.24.1",
     "chai": "^3.5.0",
-    "cross-env": "^3.1.4",
+    "cross-env": "^4.0.0",
     "devtron": "^1.4.0",
-    "electron": "1.6.2",
-    "electron-builder": "^14.5.3",
-    "electron-builder-squirrel-windows": "^15.0.0",
+    "electron": "1.6.6",
```
Comment (jasonblais) [inline package.json]: Do we want to wait for a bit until we update to 1.6.8? That version of Electron was needed for this PR https://github.com/mattermost/desktop/issues/341

Comment (yuya-oc) [inline package.json]: 1.6.8 was still beta when I created this PR. And 1.6.7 has important changes for Windows. 

- Desktop notification for Windows 7
- Per-monitor DPI awareness for Windows 8.1/10

So I want to take 1.6.6 for now.

Comment (jasonblais) [inline package.json]: Sounds good 👍

**Comment by yuya-oc** [comment]
Thanks for testing! I will check #537

**Pull request merged**